language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/util/FieldAccessCompletionResolver.java

```diff
@@ -267,24 +275,45 @@ private Symbol getSymbolByName(List<Symbol> visibleSymbols, String name, @Nonnul
         return visibleEntries;
     }
 
-    private boolean withValidAccessModifiers(Node exprNode, MethodSymbol methodSymbol, Package currentPackage,
+    private boolean withValidAccessModifiers(Node exprNode, Symbol symbol, Package currentPackage,
                                              ModuleId currentModule) {
         Optional<Project> project = context.workspace().project(context.filePath());
-        Optional<ModuleSymbol> methodSymbolModule = methodSymbol.getModule();
-        if (project.isEmpty() || methodSymbolModule.isEmpty()) {
+        Optional<ModuleSymbol> symbolModule = symbol.getModule();
+        if (project.isEmpty() || symbolModule.isEmpty()) {
             return false;
         }
-        boolean isResource = methodSymbol.qualifiers().contains(Qualifier.RESOURCE);
-        boolean isPrivate = methodSymbol.qualifiers().contains(Qualifier.PRIVATE);
-        boolean isPublic = methodSymbol.qualifiers().contains(Qualifier.PUBLIC);
+
+        boolean isPrivate = false;
+        boolean isPublic = false;
+        boolean isResource = false;
+
+        switch (symbol.kind()) {
+            case CLASS_FIELD:
```

Comment:
Can't we simplify this? I think we can keep a list to maintain the qualifiers list and then do a ingle contains logic at the end.

Reply:
Here we are not able to use the contains method as we have to use extended logic to decide the visibility based on the context.